### PR TITLE
fix(time-signature): add irrational time signatures

### LINKED ISSUE #348: error when trying to get TimeSignature with lower > 9
> TimeSignature.get('12/10')
Uncaught TypeError: Cannot read properties of undefined (reading 'split')
    at parse (@tonaljs/time-signature/dist/index.js:62:19)
    at parse (@tonaljs/time-signature/dist/index.js:55:12)
    at Object.get (@tonaljs/time-signature/dist/index.js:48:20)

### PR DESCRIPTION
**Pull request by manniL**
Resolves #348

### CODE REVIEW

File: packages/time-signature/test.ts [hunk 1]
```diff
@@ -34,6 +34,17 @@ describe("time-signature", () => {
     expect(TimeSignature.get("3+2+2/8").type).toEqual("irregular");
   });
 
+  test("get lower >= 10", () => {
+    expect(TimeSignature.get("12/10")).toEqual({
+      additive: [],
+      empty: false,
+      lower: 10,
+      name: "12/10",
+      type: "irregular",
```
Comment (danigb) [inline packages/time-signature/test.ts:43]: As I mentioned in the issue, this is an `irrational` time signature.

File: packages/time-signature/test.ts [hunk 2]
```diff
@@ -34,6 +34,17 @@ describe("time-signature", () => {
     expect(TimeSignature.get("3+2+2/8").type).toEqual("irregular");
   });
 
+  test("get lower >= 10", () => {
```
Comment (danigb) [inline packages/time-signature/test.ts:37]: I guess we could change the test name...

```suggestion
  test("irrational time signatures", () => {
```

**Comment by manniL** [comment]
According to #348, the implementation is a bit more difficult 😄

**Review by danigb — CHANGES_REQUESTED** [review]
Thanks for the PR! A couple of things:

- It would be nice to add some documentation at the README
- As you can probably [read](https://github.com/tonaljs/tonal/issues/348#issuecomment-1329462394) I was reluctant to add this kind of signatures... The problem that I see is that now any fractional can be parsed as a time signature, and I'm not sure (I don't know about irrational time signatures enough) that all them are valid time signatures... So I'd like to find (if any) an example of a valid fraction that is not a valid time signature, and add it to the tests... For example, is `123/231934` a valid irrational time signature? I don't know... (but my music theory intuition says no)

**Comment by manniL** [comment]
Valid points! Sadly, my music theory knowledge is *very* limited and I don't know much about irrational time signatures either 🙈

The question is - are there even "invalid" time signatures except having a 0 as upper/lower?

**Review by danigb — APPROVED** [review]
Great! 🚀  Thanks! 🙌

**Pull request merged**